dev-libs/hyprlang/hyprlang-0.6.3.ebuild

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake toolchain-funcs
+
+DESCRIPTION="Official implementation library for the hypr config language"
+HOMEPAGE="https://github.com/hyprwm/hyprlang"
+SRC_URI="https://github.com/hyprwm/${PN}/archive/v${PV}.tar.gz -> ${P}.gh.tar.gz"
+
+LICENSE="LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND=">=gui-libs/hyprutils-0.7.1:="
+DEPEND="${RDEPEND}"
+BDEPEND="|| ( >=sys-devel/gcc-14:* >=llvm-core/clang-18:* )"
+
+pkg_setup() {
+	[[ ${MERGE_TYPE} == binary ]] && return
+
+	if tc-is-gcc && ver_test $(gcc-version) -lt 14; then
+		eerror ">=sys-devel/gcc-14 is required to build"
+		eerror "Please upgrade GCC: emerge -v1 sys-devel/gcc"
+		die "GCC version is too old to compile!"
+	elif tc-is-clang && ver_test $(clang-version) -lt 18; then
+		eerror ">=llvm-core/clang-18 is required to build"
+		eerror "Please upgrade Clang: emerge -v1 llvm-core/clang"
+		die "Clang version is too old to compile!"
+	fi
+}
```

Comment:
Please check out `tc-check-min_ver()` from toolchain-functs.eclass.

I introduced it in commit 10b4b0976610ae256e6f3bba5181ca57dfad0e08 as a replacement for `KDE_GCC_MINIMAL` of old.

@thesamesam ^

Reply:
I'd forgot about this, I'll port Botan too..

Reply:
Thanks you. Neat helper function.